internal/users/cache/update.go

```diff
@@ -63,6 +63,11 @@ func updateUser(buckets map[string]bucketWithName, userContent userDB) {
 		_ = buckets[userByNameBucketName].Delete([]byte(existingUser.Name)) // No error as we are not in a RO transaction.
 	}
 
+	// Ensure that we use the same homedir as the one we have in cache.
+	if existingUser.Dir != "" {
+		userContent.Dir = existingUser.Dir
```

Comment:
maybe we should issue a warning still for local debugging?

Reply:
I would be fully explicit and write "instead of …".
Also, here, you spawn the warning everytime we update the user, while I think it should be when the home don’t match.

Reply:
Makes sense. I'll refine it.